src/BizHawk.Client.EmuHawk/tools/Lua/LuaConsole.cs

```diff
@@ -1388,18 +1388,14 @@ private void InputBox_KeyDown(object sender, KeyEventArgs e)
 
 					LuaSandbox.Sandbox(null, () =>
 					{
-						LuaImp.ExecuteString($"console.log({rawCommand})");
-					}, () =>
-					{
-						LuaSandbox.Sandbox(null, () =>
+						var results = LuaImp.ExecuteString(rawCommand);
+						// empty array if the command was e.g. a variable assignment or a loop without return statement
+						// "void" functions return a single null
+						// if output didn't change, Print will take care of writing out "(no return)"
+						if (results is not [ ] and not [ null ] || OutputBox.Text == consoleBeforeCall)

```

Comment:
nitpick: `is not ([] or [ null ])`
also `ReferenceEquals` for string comparison maybe

Reply:
ReferenceEquals should not be used. The getter for Output.Text has no guarantee to actually be the same reference with 2 calls to it (especially in this case, getting the text from a native window is going to be a p/invoke call, which means creating a new string, i.e. a new reference)